javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,84 @@
+let name = "";
+let toDoList = [];
+
+function getReply(command){
+    // lowercase command and split it into commandWords array
+   const commandWords = command.toLowerCase().split(" ");
+   let response = null;
+   let toDoText = "";
+
+   if (name !== "" && commandWords[0] == "hello"){
+    console.log("you've already introduced yourself") 
+    return 
+   }
+
+   switch(commandWords[0]) {
+      case "hello":
+        name = commandWords.pop();
+        response = "nice to meet you " + name; break;
+
+      case "what":
+        switch(true) {
+            case commandWords.includes("name"):{
+              if (name === ""){
+                response = "sorry! you have not told us your name yet!"; break;
+              }
+              response = `your name is ${name} `; break;}
+            case commandWords.includes("todo?"):
+              response = toDoList; break;
+            case commandWords.includes("today?"):
+              response = new Date().toDateString(); break;
+            case commandWords[3] == "+":
+              response = parseInt(commandWords[2]) + parseInt(commandWords[4]); break;
+            case commandWords[3] == "*":
+              response = commandWords[2] * commandWords[4]; break;
+            case commandWords[3] == "/":
+              response = commandWords[2] / commandWords[4]; break;
+            case commandWords[3] == "-":
+              response = commandWords[2] - commandWords[4]; break;
```

Comment:
Switch-cases need a default case.

Reply:
but it is optional: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
You're right, it is! I didn't actually know that. 
But I think it's good practice to have a fallback value :)